### ISSUE TITLE
Attempting to lower land without enough money turns on underground view

### ISSUE THREAD

**Issue by Derpercoaster** [OpenLoco/OpenLoco#2560]
**Version information:**
 - Platform: Windows 10
 - Version: 24.07
 - Build: d5608cc2

**Describe the bug**
If you attempt to lower the land when you don't have enough money to do so, underground view turns itself on.

**To Reproduce**
Steps to reproduce the behaviour:
1. Be in a scenario
2. Have too little money to lower land
3. Attempt to lower land
4. Underground view will turn on when the "not enough money" error pops up

**Expected behaviour**
Underground view should not turn on when attempting to lower land

**Additional context**
I'm going to assume the game _normally_ compares the 'old' land height value (what the height was) to the 'new' land height value (what you just changed it to), which is how it changes the height for you to see. Except in this case, it is erroneously looking at the 'desired' height (what you want it to change to) instead of the 'new' height. Since the desired height is lower than the current height, it turns on underground view, as it assumes you're working underground.

In other words, I think the game is looking at the wrong value when it refuses to change the land height (on account of you being poor), thus causing the bug. So in a way, it is technically correct that you are now underground, but in _the_ most technical manner I've ever seen. Either that or I am looking too deeply into this.

**Comment by LeftofZen** [comment]
@Derpercoaster is this still a bug for you? I just tried in 24.10 and I don't get underground view turned on
